### PR TITLE
[CBRD-25139] When using oracle style empty string, the result of replace is null.

### PR DESCRIPTION
**Pull request by beyondykk9**
http://jira.cubrid.org/browse/CBRD-25139

When setting oracle style empty string=y, the result is null when using syntax such as replace(?,'-','').

CREATE TABLE tbl(i VARCHAR(100));
INSERT INTO tbl SELECT '2013-01-01' FROM dual;
SELECT  REPLACE(i,'-','') FROM tbl;

Expected Result
```
  replace(i, '-', '')
======================
  '20130101' 
```

Actual Result:
```
  replace(i, '-', '')
======================
  NULL
```

### CODE REVIEW

File: src/parser/type_checking.c [hunk 1]
```diff
@@ -18237,17 +18237,31 @@ pt_fold_const_expr (PARSER_CONTEXT * parser, PT_NODE * expr, void *arg)
...
       if (domain
 	  && (QSTR_IS_ANY_CHAR_OR_BIT (TP_DOMAIN_TYPE (domain)) || type1 == PT_TYPE_NULL || type2 == PT_TYPE_NULL))
 	{
-	  if (opd1 && opd1->node_type == PT_VALUE && type1 == PT_TYPE_NULL)
+	  if (opd3 && opd3->node_type == PT_VALUE && type3 == PT_TYPE_NULL)	/* REPLACE */
+	    {
+	      result = parser_copy_tree (parser, opd1);
+	      if (result == NULL)
+		{
+		  has_error = true;
```
Comment (hyunikn) [inline src/parser/type_checking.c]: no 'goto end;' after this line?

Comment (beyondykk9) [reply src/parser/type_checking.c]: Good point.

File: src/parser/type_checking.c [hunk 2]
```diff
@@ -18237,24 +18237,37 @@ pt_fold_const_expr (PARSER_CONTEXT * parser, PT_NODE * expr, void *arg)
...
+	}
+
       /* use the caching variant of this function ! */
       domain = pt_xasl_node_to_domain (parser, expr);
 
       if (domain
 	  && (QSTR_IS_ANY_CHAR_OR_BIT (TP_DOMAIN_TYPE (domain)) || type1 == PT_TYPE_NULL || type2 == PT_TYPE_NULL))
 	{
-	  if (opd1 && opd1->node_type == PT_VALUE && type1 == PT_TYPE_NULL)
+	  if (opd3 && opd3->node_type == PT_VALUE && type3 == PT_TYPE_NULL)	/* REPLACE */
```
Comment (hyunikn) [inline src/parser/type_checking.c]: Adding this conditional block results in 
```
csql> select replace('abc', 'a', null);

=== <Result of SELECT Command in Line 1> ===

   replace('abc', 'a', null)
======================
  'abc'
```
while the current result is 
```
csql> select replace('abc', 'a', null);

=== <Result of SELECT Command in Line 1> ===

   replace('abc', 'a', null)
======================
  NULL                
```
Note the Oracle's result of 
```
SELECT replace('abc', 'a', null) FROM dual;
```
is 'bc'.

Comment (beyondykk9) [reply src/parser/type_checking.c]: Good point, I fixed it.

File: src/parser/type_checking.c [hunk 3]
```diff
@@ -18237,24 +18237,44 @@ pt_fold_const_expr (PARSER_CONTEXT * parser, PT_NODE * expr, void *arg)
...
+	      if (err != NO_ERROR)
+		{
+		  has_error = true;
+		}
+	      else
+		{
+		  result = expr;
+		}
+	    }
+	  else if (opd1 && opd1->node_type == PT_VALUE && type1 == PT_TYPE_NULL)
```
Comment (hyunikn) [inline src/parser/type_checking.c:18274]: "select replace(null, 'a', 'A');" results in a wrong value 'a' due to this conditional block.
Separate handling of REPLACE seems necessary.

File: src/query/fetch.c [hunk 4]
```diff
@@ -1551,7 +1551,8 @@ fetch_peek_arith (THREAD_ENTRY * thread_p, REGU_VARIABLE * regu_var, val_descr *
 	      goto error;
 	    }
 	}
-      if (DB_IS_NULL (peek_left) || (peek_third && DB_IS_NULL (peek_third)))
+      if (!prm_get_bool_value (PRM_ID_ORACLE_STYLE_EMPTY_STRING)
+	  && (DB_IS_NULL (peek_left) || (peek_third && DB_IS_NULL (peek_third))))
```
Comment (hyunikn) [inline src/query/fetch.c]:
```suggestion
	  && (DB_IS_NULL (peek_left) || DB_IS_NULL (peek_right) || (peek_third && DB_IS_NULL (peek_third))))
```
just for readability.

File: src/parser/type_checking.c (hunk 3 above)
Comment (beyondykk9) [reply src/parser/type_checking.c:18274]: Good point, I fixed it.

**Review by ctshim — APPROVED** [review]

**Review by hyunikn — APPROVED** [review]

**Pull request merged**